<!DOCTYPE html>
<html lang="en-US">
  <head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">

<!-- Begin Jekyll SEO tag v2.8.0 -->
<title>SEtoolbox</title>
<meta name="generator" content="Jekyll v3.10.0" />
<meta property="og:title" content="SEtoolbox" />
<meta property="og:locale" content="en_US" />
<link rel="canonical" href="http://localhost:4000/" />
<meta property="og:url" content="http://localhost:4000/" />
<meta property="og:site_name" content="SEtoolbox" />
<meta property="og:type" content="website" />
<meta name="twitter:card" content="summary" />
<meta property="twitter:title" content="SEtoolbox" />
<script type="application/ld+json">
{"@context":"https://schema.org","@type":"WebSite","headline":"SEtoolbox","name":"SEtoolbox","url":"http://localhost:4000/"}</script>
<!-- End Jekyll SEO tag -->

    <link rel="stylesheet" href="/assets/css/style.css?v=e0416f889c9aa2f7ec7044684760e5a31dcc9b68">
    <!-- start custom head snippets, customize with your own _includes/head-custom.html file -->

<!-- Setup Google Analytics -->



<!-- You can set your favicon here -->
<!-- link rel="shortcut icon" type="image/x-icon" href="/favicon.ico" -->

<!-- end custom head snippets -->

  </head>
  <body>
    <div class="container-lg px-3 my-5 markdown-body">
      
      <h1><a href="http://localhost:4000/">SEtoolbox</a></h1>
      

      
<ul>
  <li>
    <p><strong>Shaoxun Yuan</strong></p>
  </li>
  <li>
    <p><strong>Affiliation</strong>: School of Artificial Intelligence and Information Technology, Nanjing University of Chinese Medicine, China</p>
  </li>
  <li>
    <p><strong>Email</strong>: <a href="mailto:yuanshaoxun@njucm.edu.cn">yuanshaoxun@njucm.edu.cn</a></p>
  </li>
</ul>

<div class="toc">  
    <ul>  
          
              
          
    </ul>  
</div>

<h2 id="1-introduction">1. Introduction</h2>

<p>SEtoolbox is an R package that operates, analyzes and visualizes SummarizedExperiment objects.</p>

<h2 id="2-installation">2. Installation</h2>

<p>To install the SEtoolbox package, you first need to install the <code class="language-plaintext highlighter-rouge">devtools</code> package.</p>

<p>Run the following command in your R console:</p>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">install.packages</span><span class="p">(</span><span class="s2">"devtools"</span><span class="p">)</span><span class="w">

</span><span class="n">devtools</span><span class="o">::</span><span class="n">install_github</span><span class="p">(</span><span class="s2">"shaoxunyuan/SEtoolbox"</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<h2 id="3-load-packages-required-for-this-tutorial">3. Load packages required for this tutorial</h2>

<p>During this tutorial, we might need to use a few additional packages.</p>

<p>Since we specified dependencies = TRUE when installing SEtoolbox package, these additional packages have already been installed.</p>

<p>We can load them directly.</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">library</span><span class="p">(</span><span class="n">SummarizedExperiment</span><span class="p">)</span><span class="w">

</span><span class="n">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span><span class="w">

</span><span class="n">library</span><span class="p">(</span><span class="n">plyr</span><span class="p">)</span><span class="w">

</span><span class="n">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span><span class="w">

</span><span class="n">library</span><span class="p">(</span><span class="n">reshape2</span><span class="p">)</span><span class="w">

</span><span class="n">library</span><span class="p">(</span><span class="n">DESeq2</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<h2 id="4-input-data">4. Input data</h2>

<p>For this tutorial, SEtoolbox will be working with a <a href="https://www.bioconductor.org/packages/devel/bioc/vignettes/SummarizedExperiment/inst/doc/SummarizedExperiment.html"><code class="language-plaintext highlighter-rouge">SummarizedExperiment</code></a> object.</p>

<h2 id="5-functions">5. Functions</h2>

<p>Functions in SEtoolbox can be obtain using</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">help</span><span class="p">(</span><span class="n">package</span><span class="o">=</span><span class="s2">"SEtoolbox"</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<h3 id="51-se_combine">5.1. SE_combine</h3>

<p>Combine multiple <code class="language-plaintext highlighter-rouge">SummarizedExperiment</code> objects.</p>

<p>This function merges multiple <code class="language-plaintext highlighter-rouge">SummarizedExperiment</code> objects based on the specified merge type (intersection or union) for all assays present in the input list.</p>

<h4 id="usage">Usage</h4>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>SE_combine(se_list, merge_type = "intersection") 
</code></pre></div></div>

<h4 id="arguments">Arguments</h4>

<p><code class="language-plaintext highlighter-rouge">se_list</code>: A list of SummarizedExperiment objects to be combined (se_list = list(SE1,SE2,SE3))</p>

<p><code class="language-plaintext highlighter-rouge">merge_type</code>: A character string specifying the type of merge to perform. Options are:</p>

<p>    1.<code class="language-plaintext highlighter-rouge">intersection</code> (default): Keep only common features across all objects.</p>

<p>    2.<code class="language-plaintext highlighter-rouge">union</code>: Keep all features.</p>

<h3 id="52-se_impute">5.2 SE_impute</h3>

<p>Fill missing values in a <code class="language-plaintext highlighter-rouge">SummarizedExperiment</code> object.</p>

<p>This function imputes missing values (NA) in the given <code class="language-plaintext highlighter-rouge">SummarizedExperiment</code> object using specified methods.</p>

<p>Multiple imputation techniques can be utilized to handle missing values, ensuring the robustness of subsequent analyses.</p>

<h4 id="usage-1">Usage</h4>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>SE_impute(object, assayname = "TPM", group = "group", ZerosAsNA = FALSE, RemoveNA = TRUE,  
          cutoff = 20, method = c("none", "LOD", "half_min", "median", "mean", "min", "knn", "rf", "global_mean", "svd", "QRILC"),  
          LOD = NULL, knum = 10)  
</code></pre></div></div>

<h4 id="arguments-1">Arguments</h4>

<p><code class="language-plaintext highlighter-rouge">object</code>: A <code class="language-plaintext highlighter-rouge">SummarizedExperiment</code> object containing the data to be imputed.</p>

<p><code class="language-plaintext highlighter-rouge">assayname</code>: The name of the <code class="language-plaintext highlighter-rouge">SummarizedExperiment</code> assay, specifying the type of data to be imputed.</p>

<p><code class="language-plaintext highlighter-rouge">group</code>: A character string specifying the grouping variable in the sample data.</p>

<p><code class="language-plaintext highlighter-rouge">ZerosAsN</code>: A logical value indicating whether to treat zeros as NA. Default is <code class="language-plaintext highlighter-rouge">FALSE</code>.</p>

<p><code class="language-plaintext highlighter-rouge">RemoveNA</code>: A logical value indicating whether to remove samples with a high percentage of NA values based on the cutoff. Default is <code class="language-plaintext highlighter-rouge">TRUE</code>.</p>

<p><code class="language-plaintext highlighter-rouge">cutoff</code>: A numerical value representing the percentage cutoff for NA samples. Default is <code class="language-plaintext highlighter-rouge">20</code>.</p>

<p><code class="language-plaintext highlighter-rouge">method</code>: A character string specifying the imputation method to use. Options include:</p>

<p>    1.<code class="language-plaintext highlighter-rouge">none</code> (default): No imputation, replace NA with zero.</p>

<p>    2.<code class="language-plaintext highlighter-rouge">LOD</code>: Replace NA with the limit of detection (LOD).</p>

<p>    3.<code class="language-plaintext highlighter-rouge">half_min</code>: Replace NA with half of the minimum value.</p>

<p>    4.<code class="language-plaintext highlighter-rouge">median</code>: Replace NA with the median value.</p>

<p>    5.<code class="language-plaintext highlighter-rouge">mean</code>: Replace NA with the mean value.</p>

<p>    6.<code class="language-plaintext highlighter-rouge">min</code>: Replace NA with the minimum value.</p>

<p>    7.<code class="language-plaintext highlighter-rouge">knn</code>: K-nearest neighbors imputation.</p>

<p>    8.<code class="language-plaintext highlighter-rouge">rf</code>: Random forest imputation.</p>

<p>    9.<code class="language-plaintext highlighter-rouge">global_mean</code>: Global mean imputation.</p>

<p>    10.<code class="language-plaintext highlighter-rouge">svd</code>: Singular value decomposition imputation.</p>

<p>    11.<code class="language-plaintext highlighter-rouge">QRILC</code>: Quantile regression imputation.</p>

<p><code class="language-plaintext highlighter-rouge">LOD</code>: A numerical value representing the limit of detection (used for the LOD imputation method). Default is <code class="language-plaintext highlighter-rouge">NULL</code>.</p>

<p><code class="language-plaintext highlighter-rouge">knum</code>: An integer value representing the number of neighbors in the KNN imputation method. Default is <code class="language-plaintext highlighter-rouge">10</code>.</p>

<h3 id="53-se_detectraio">5.3 SE_detectraio</h3>

<p>Calculate detection ratio and update <code class="language-plaintext highlighter-rouge">SummarizedExperiment</code> object’s <code class="language-plaintext highlighter-rouge">rowData</code>.</p>

<p>This function computes the detection ratio of expression data and updates the <code class="language-plaintext highlighter-rouge">rowData</code> of the provided <code class="language-plaintext highlighter-rouge">SummarizedExperiment</code> object with detection sample counts and ratios. It also generates a histogram of detection ratios.</p>

<h4 id="usage-2">Usage</h4>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>SE_detectratio(SE, assayname = "TPM")  
</code></pre></div></div>

<h4 id="arguments-2">Arguments</h4>

<p><code class="language-plaintext highlighter-rouge">SE</code>: A <code class="language-plaintext highlighter-rouge">SummarizedExperiment</code> object containing expression data.</p>

<p><code class="language-plaintext highlighter-rouge">assayname</code>: The name of the assay to be used for calculations. Default is <code class="language-plaintext highlighter-rouge">"TPM"</code>.</p>

<h3 id="54-se_deseq2">5.4 SE_DEseq2</h3>

<p>Perform differential expression analysis using <code class="language-plaintext highlighter-rouge">DESeq2</code>.</p>

<p>This function performs differential expression analysis on count data contained in a <code class="language-plaintext highlighter-rouge">SummarizedExperiment</code> object using the <code class="language-plaintext highlighter-rouge">DESeq2</code> package.</p>

<h4 id="usage-3">Usage</h4>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>SE_DEseq2(SE, assayname = "Count", groupname = "group")  
</code></pre></div></div>

<h4 id="arguments-3">Arguments</h4>

<p><code class="language-plaintext highlighter-rouge">SE</code>: A <code class="language-plaintext highlighter-rouge">SummarizedExperiment</code> object containing count data.</p>

<p><code class="language-plaintext highlighter-rouge">assayname</code>: The name of the assay to use for the analysis. Default is <code class="language-plaintext highlighter-rouge">"Count"</code>.</p>

<p><code class="language-plaintext highlighter-rouge">groupname</code>: The name of the column in <code class="language-plaintext highlighter-rouge">colData(SE)</code> that contains the factor for grouping samples. Default is <code class="language-plaintext highlighter-rouge">"group"</code>.</p>

<h3 id="55-se_boxplot">5.5 SE_boxplot</h3>

<p>Generates a boxplot for specified features in a <code class="language-plaintext highlighter-rouge">SummarizedExperiment</code> object.</p>

<p>This function creates a boxplot for the specified features (genes) within a given <code class="language-plaintext highlighter-rouge">SummarizedExperiment</code> object.</p>

<p>It supports normalization and grouping of the data.</p>

<h4 id="usage-4">Usage</h4>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>SE_boxplot(SE, feature_of_interest, assayname = "TPM", group_col = NA, normalization = "none")  
</code></pre></div></div>

<h4 id="arguments-4">Arguments</h4>

<p><code class="language-plaintext highlighter-rouge">SE</code>: A SummarizedExperiment object.</p>

<p><code class="language-plaintext highlighter-rouge">feature_of_interest</code>: Character vector of gene identifiers.</p>

<p><code class="language-plaintext highlighter-rouge">assayname</code>: The assay name in the SummarizedExperiment object.</p>

<p><code class="language-plaintext highlighter-rouge">group_col</code>: Column name in colData(SE) for grouping.</p>

<p><code class="language-plaintext highlighter-rouge">normalization</code>: Normalization method (“scale”, “log”, or “none”).</p>

<h3 id="56-se_pcaplot">5.6 SE_PCAplot</h3>

<p>Generate PCA plots.</p>

<p>This function takes a <code class="language-plaintext highlighter-rouge">SummarizedExperiment</code> object, computes PCA, and visualizes the results.</p>

<h4 id="usage-5">Usage</h4>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>SE_PCAplot(SE, assayname = "TPM", groupname = "group", outlier_threshold = 2, scale = TRUE, feature_of_interesting = NULL)  
</code></pre></div></div>

<h4 id="arguments-5">Arguments</h4>

<p><code class="language-plaintext highlighter-rouge">SE</code>: SummarizedExperiment object containing gene expression data.</p>

<p><code class="language-plaintext highlighter-rouge">assayname</code>: Name of the expression data, default is “TPM”.</p>

<p><code class="language-plaintext highlighter-rouge">groupname</code>: Name of the grouping column, default is “group”.</p>

<p><code class="language-plaintext highlighter-rouge">outlier_threshold</code>: Outlier filtering threshold, default is 2.</p>

<p><code class="language-plaintext highlighter-rouge">scale</code>: Whether to standardize the data, default is TRUE.</p>

<p><code class="language-plaintext highlighter-rouge">feature_of_interesting</code>: Vector of specific feature names; if NULL, all features are used, default is NULL.</p>

<h2 id="6-acknowledgements">6. Acknowledgements</h2>

<p>The author would like to thank Deepseek.</p>

<h2 id="7-session-info">7. Session Info</h2>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>&gt; sessionInfo()
R version 4.2.0 (2022-04-22)
Platform: x86_64-conda-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /home/shaoxun/anaconda3/envs/yuanshaoxun/lib/libopenblasp-r0.3.21.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C
 [9] LC_ADDRESS=C               LC_TELEPHONE=C
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base

other attached packages:
[1] SEtoolbox_0.1.0

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.14                 pillar_1.10.1
 [3] bslib_0.9.0                 compiler_4.2.0
 [5] later_1.4.1                 jquerylib_0.1.4
 [7] GenomeInfoDb_1.34.9         XVector_0.38.0
 [9] MatrixGenerics_1.10.0       bitops_1.0-9
[11] tools_4.2.0                 zlibbioc_1.44.0
[13] digest_0.6.37               tibble_3.2.1
[15] gtable_0.3.6                lattice_0.22-6
[17] jsonlite_1.9.1              memoise_2.0.1
[19] lifecycle_1.0.4             pkgconfig_2.0.3
[21] rlang_1.1.5                 Matrix_1.6-5
[23] DelayedArray_0.24.0         shiny_1.10.0
[25] cli_3.6.4.9000              fastmap_1.2.0
[27] GenomeInfoDbData_1.2.9      dplyr_1.1.4
[29] generics_0.1.3              vctrs_0.6.5
[31] S4Vectors_0.36.2            sass_0.4.9
[33] IRanges_2.32.0              tidyselect_1.2.1
[35] grid_4.2.0                  stats4_4.2.0
[37] glue_1.8.0                  Biobase_2.58.0
[39] R6_2.6.1                    ggplot2_3.5.1
[41] magrittr_2.0.3              scales_1.3.0
[43] promises_1.3.2              matrixStats_1.5.0
[45] htmltools_0.5.8.1           BiocGenerics_0.44.0
[47] GenomicRanges_1.50.2        SummarizedExperiment_1.28.0
[49] colorspace_2.1-1            mime_0.12
[51] xtable_1.8-4                httpuv_1.6.15
[53] munsell_0.5.1               RCurl_1.98-1.16
[55] cachem_1.1.0
</code></pre></div></div>

<h2 id="8-references">8. References</h2>

<p>Will update soon</p>


      
      <div class="footer border-top border-gray-light mt-5 pt-3 text-right text-gray">
        This site is open source. <a href="https://github.com/shaoxunyuan/SEtoolbox/edit/gh-pages/index.md">Improve this page</a>.
      </div>
      
    </div>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/anchor-js/4.1.0/anchor.min.js" integrity="sha256-lZaRhKri35AyJSypXXs4o6OPFTbTmUoltBbDCbdzegg=" crossorigin="anonymous"></script>
    <script>anchors.add();</script>
  </body>
</html>
